hpx/util/ini.hpp

```diff
@@ -93,43 +124,117 @@ namespace hpx { namespace util
         void dump(int ind = 0, std::ostream& strm = std::cout) const;
 
         void add_section(std::string const& sec_name, section& sec,
-            section* root = nullptr);
-        bool has_section(std::string const& sec_name) const;
+            section* root = nullptr)
+        {
+            util::ignore_mutex_while_checking<mutex_type> im(mtx_);
```

Comment:
Hmmm, now as I think of it, we might get away without ignoring it. I'll play with this for a bit...